velox/connectors/hive/storage_adapters/s3fs/CMakeLists.txt

```diff
@@ -19,7 +19,11 @@ if(VELOX_ENABLE_S3)
   target_sources(velox_s3fs PRIVATE S3FileSystem.cpp S3Util.cpp)
 
   target_include_directories(velox_s3fs PUBLIC ${AWSSDK_INCLUDE_DIRS})
-  target_link_libraries(velox_s3fs Folly::folly ${AWSSDK_LIBRARIES})
+  target_link_libraries(velox_s3fs Folly::folly ${AWSSDK_LIBRARIES} xsimd)
```

Comment:
Why need s3 related changes?

Reply:
@JkSelf Thank you for your reply.
Because we included `velox/dwio/common/DataBuffer.h` in the `velox/common/file/File.h` file, and `DataBuffer.h` depends on `xsimd` and `gtest`

Reply:
Why does `DataBuffer.h` depend on xsimd and gtest? We need to fix this.

Reply:
The `CMakeLists.txt` file of s3 can not be modified this time, but the `CMakeLists.txt` file of hdfs depends on this this time.
if we remove xsimd The following exception occurs at compile time:
```
In file included from /Users/wypb/data/code/apache/temp/velox/velox/connectors/hive/storage_adapters/hdfs/RegisterHdfsFileSystem.cpp:23:
In file included from /Users/wypb/data/code/apache/temp/velox/./velox/dwio/common/DataSink.h:23:
In file included from /Users/wypb/data/code/apache/temp/velox/./velox/dwio/common/DataBuffer.h:22:
In file included from /Users/wypb/data/code/apache/temp/velox/./velox/buffer/Buffer.h:26:
/Users/wypb/data/code/apache/temp/velox/./velox/common/base/SimdUtil.h:24:10: fatal error: 'xsimd/xsimd.hpp' file not found
#include <xsimd/xsimd.hpp>
```
if remove gtest, The following exception occurs at compile time:
```
In file included from /Users/wypb/data/code/apache/temp/velox/velox/connectors/hive/storage_adapters/hdfs/RegisterHdfsFileSystem.cpp:23:
In file included from /Users/wypb/data/code/apache/temp/velox/./velox/dwio/common/DataSink.h:23:
In file included from /Users/wypb/data/code/apache/temp/velox/./velox/dwio/common/DataBuffer.h:22:
In file included from /Users/wypb/data/code/apache/temp/velox/./velox/buffer/Buffer.h:27:
In file included from /Users/wypb/data/code/apache/temp/velox/./velox/common/memory/Memory.h:38:
/Users/wypb/data/code/apache/temp/velox/./velox/common/base/GTestMacros.h:24:10: fatal error: 'gtest/gtest_prod.h' file not found
#include <gtest/gtest_prod.h>
         ^~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

Reply:
Including `velox/common/base/GTestMacros.h` inside `velox/common/memory/Memory.h` seems unnecessary.
I added a PR https://github.com/facebookincubator/velox/pull/5930/files to remove it. Let's see if something breaks.